### ISSUE TITLE
Two try/catch blocks in a row unsupported?

### ISSUE THREAD

**Issue by michal-grzejszczak** [MatAtBread/nodent#4]
The following code compiles to incorrect JavaScript. The second try/catch blocks ends up as a $post_try_$1 function that refers to not existing reference $error.

Is that a known limitation?

```
async function doStuff() {
    setTimeout(function() {
        throw async new Error("error");
    }, 1e3);
};

try {
    await doStuff();
} catch (e) {
    console.log("caught error");
}

try {
    await doStuff();
} catch (e) {
    console.log("caught error");
}
```

**Comment by matAtWork** [comment]
Can you tell me which version you're compiling with? There were some issue I fixed related to try-catch blocks

**Comment by matAtWork** [comment]
Looks like a bug - thanks for pointing it out. I'll generate a test case and find a fix.

**Comment by matAtWork** [comment]
Good catch (no pun intended). The issue is that the synthetic $post() continuation function was hoisted to its parent scope (the function) to fix an issue with Firefox not liking forward references to functions. 

By hoisting it to the the top of it's scope, it was hoisted out of the try block (and therefore the catch). I'll modify the logic to hoist to the front of the nearest function scope OR try block (whichever is nearest) and check that works against the existing tests.

**Comment by michal-grzejszczak** [comment]
It took me a while to grasp the "good catch" part :) I was using latest version from npm as well as online demo, but I see that you already know it.

**Comment by matAtWork** [comment]
I'll try and get a fix out in the next day or so. Can't believe I've not seen this one before (duh)

**Comment by matAtWork** [comment]
I've just pushed to the 'dev' branch a test (tests/try-pair.js) and a fix. All seems good, but I'll do a bit more testing before publishing to npm. If you could verify it fixes your issue, that would be great.

**Comment by MatAtBread** [comment]
Fixed as of version 1.1.2
